spring-integration-graphql/src/main/java/org/springframework/integration/graphql/outbound/GraphQlMessageHandler.java

```diff
@@ -100,6 +107,23 @@ public void setVariablesExpression(Expression variablesExpression) {
 		this.variablesExpression = variablesExpression;
 	}
 
+	/**
+	 * Set a Locale for GraphQL Query to execute.
+	 * @param locale the locale to use.
+	 */
+	public void setLocale(@Nullable Locale locale) {
+		this.locale = locale;
+	}
+
+	/**
+	 * Set a Execution Id for GraphQL Query to execute.
+	 * @param executionId the executionId to use.
+	 */
+	public void setExecutionId(String executionId) {
```

Comment:
Well, maybe exactly this `executionId` should be evaluated really against request message.
Therefore an expression. 

I don't know details of this property, so what is your take ?

Thanks

Reply:
`executionId` should be unique per request. There is a GraphQL Java `ExecutionIdProvider` that could be set on `graphql.GraphQL` and by default there is one that creates a random UUID. However it's also possible to set the `executionId` directly on `ExecutionInput` for each request. The Spring MVC and WebFlux transports set it to an HTTP request id, while the WebSocket transports set it to the subscription id of the GraphQL over WebSocket protocol. It can also be overridden from a `WebInterceptor` by using `RequestInput#executionId`.

All in all, a `setExecutionId(String)` on `GraphQlMessageHandler`, which I think is a singleton, doesn't look right indeed. It would have to be a generator callback of some kind and then there is the question of what input do you need to determine it. This could be dropped for now until there is more clarity on the actual need.

Reply:
Thanks, Rossen!

So, yeah, my guess that it has to be unique, per-message is correct. 
Therefore we need to go with a SpEL expression over here where incoming `Message` is going to be root evaluation object. 
Similar to what we do now with that `evaluateOperationNameExpression()`.
I believe by default it can be null to let the target GraphQL engine to generate such an id for us.